File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/IndexHasChanged.py
from ravendb import IndexDefinition, IndexHasChangedOperation
from ravendb.documents.operations.definitions import MaintenanceOperation

from examples_base import ExampleBase


class IndexHasChanged(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_index_has_changed(self):
        with self.embedded_server.get_document_store("IndexHasChanged") as store:
            # region index_has_changed
            # Some index definition
            index_definition = IndexDefinition(
                name="UsersByName", maps={"from user in docs.Users select new { user.Name }"}
            )

            # Define the has-changed operation, pass the index definition
            index_has_changed_op = IndexHasChangedOperation(index_definition)

            # Execute the operation by passing it to maintenance.send
            index_has_changed = store.maintenance.send(index_has_changed_op)

            # Return values:
            # False: The definition of the index passed is the SAME as the one deployed on the server
            # True:  The definition of the index passed is DIFFERENT from the one deployed on the server
            #        Or - index does not exist
            # endregion

    class Foo:
        # region syntax
        class IndexHasChangedOperation(MaintenanceOperation[bool]):
            def __init__(self, index: IndexDefinition): ...

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/IndexHasChanged.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\IndexDefinition;
use RavenDB\Documents\Operations\Indexes\IndexHasChangedOperation;

class IndexHasChanged
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region index_has_changed
            // Some index definition
            $indexDefinition = new IndexDefinition();
            $indexDefinition->setName("UsersByName");
            $indexDefinition->setMaps(["from user in docs.Users select new { user.Name }"]);

            // Define the has-changed operation, pass the index definition
            $indexHasChangedOp = new IndexHasChangedOperation($indexDefinition);

            // Execute the operation by passing it to Maintenance.Send
            $store->maintenance()->send($indexHasChangedOp);

            // Return values:
            // false: The definition of the index passed is the SAME as the one deployed on the server
            // true:  The definition of the index passed is DIFFERENT than the one deployed on the server
            //        Or - index does not exist
            # endregion

            $store->maintenance()->send($resumeIndexingOp);
        } finally {
            $store->close();
        }
    }

}

/*
interface IFoo
{
    # region syntax
    IndexHasChangedOperation(?IndexDefinition $definition)
    # endregion
}
*/
